src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java

```diff
@@ -182,20 +268,28 @@ private void validateSchema(Schema schema) {
     }
   }
 
-  private Set<String> getCreatableSObjectFields() throws ConnectionException {
+  private Set<String> getCreatableSObjectFields(SObjectsDescribeResult describeResult) {
     Set<String> creatableSObjectFields = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
 
-    AuthenticatorCredentials credentials = this.getAuthenticatorCredentials();
-    PartnerConnection partnerConnection = new PartnerConnection(Authenticator.createConnectorConfig(credentials));
-
-    SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromName(this.getSObject(), credentials);
-    SObjectsDescribeResult describeResult = new SObjectsDescribeResult(partnerConnection,
-                                                                       sObjectDescriptor.getAllParentObjects());
     for (Field field : describeResult.getFields()) {
       if (field.isCreateable()) {
         creatableSObjectFields.add(field.getName());
       }
     }
     return creatableSObjectFields;
   }
+
+  private SObjectsDescribeResult getSObjectDescribeResult() {
+    AuthenticatorCredentials credentials = this.getAuthenticatorCredentials();
+    PartnerConnection partnerConnection = null;
+    SObjectDescriptor sObjectDescriptor = null;
+    try {
+      partnerConnection = new PartnerConnection(Authenticator.createConnectorConfig(credentials));
+      sObjectDescriptor = SObjectDescriptor.fromName(this.getSObject(),
+                                                     this.getAuthenticatorCredentials());
+    } catch (ConnectionException e) {
+      throw new InvalidStageException("There was issue communicating with Salesforce", e);
+    }
+    return new SObjectsDescribeResult(partnerConnection, sObjectDescriptor.getAllParentObjects());
```

Comment:
you don't have to return outside of try block. That means you don't have to init all variables with null. Just move all declarations and return into try block.

Reply:
done